internal/pgbackrest/options.go

```diff
@@ -0,0 +1,96 @@
+/*
+ Copyright 2021 Crunchy Data Solutions, Inc.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package pgbackrest
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// iniMultiSet represents the key-value pairs in a pgBackRest config file section.
+type iniMultiSet map[string][]string
+
+func (ms iniMultiSet) String() string {
+	keys := make([]string, 0, len(ms))
+	for k := range ms {
+		keys = append(keys, k)
+	}
+
+	sort.Strings(keys)
+
+	var b strings.Builder
+	for _, k := range keys {
+		for _, v := range ms[k] {
+			if len(v) <= 0 {
+				_, _ = fmt.Fprintf(&b, "%s =\n", k)
+			} else {
+				_, _ = fmt.Fprintf(&b, "%s = %s\n", k, v)
+			}
+		}
+	}
+	return b.String()
+}
+
+// Add associates value with key, appending it to any values already associated
+// with key. The key is case-sensitive.
+func (ms iniMultiSet) Add(key, value string) {
+	ms[key] = append(ms[key], value)
+}
+
+// Set replaces the values associated with key with the single value. The key
+// is case-sensitive.
+func (ms iniMultiSet) Set(key, value string) {
+	ms[key] = []string{value}
+}
+
+// Value returns empty string or the first value associated with the given key
+// if it is present in ms.
+func (ms iniMultiSet) Value(key string) string {
```

Comment:
:thinking: Returning the first value in this function feels a bit funny, especially if there are multiple values.  Not sure what the right answer is - maybe passing in an index or something along those lines?

Or maybe we should simply have the `Values()` function, and the client can simply inspect the slice contents as needed?

Reply:
This is for options that can only be set once. [net/http.Header](https://pkg.go.dev/net/http#Header) is a similar type; it uses "Get" for this.

Over in [postgres.ParameterSet](https://github.com/CrunchyData/postgres-operator/blob/master/internal/postgres/parameters.go#L93), we use "Value" for this.

I haven't needed to read _any_ values outside of testing, so I'll drop this method to avoid naming it. 😄

Reply:
Thanks for the reference to Header package! ... some other things in there have me thinking about naming a bit more (e.g. I see they have an `Add()` and a `Set()`).

Sounds good regarding dropping till we need it tho! :+1:

Reply:
Done in [e5f9d2c](https://github.com/CrunchyData/postgres-operator/compare/912f6c652aa820b7dc85a9e2218353b8409ab817..e5f9d2c32a83c59c31ab391fae9e94ab35d952f7).

With `Values` being the only way to read, I noticed that there was no way to set all the values for a key at once. I changed `Set` to take zero or more values.